go/vt/discovery/healthcheck.go

```diff
@@ -647,16 +649,23 @@ func (hc *HealthCheckImpl) Unsubscribe(c chan *TabletHealth) {
 	delete(hc.subscribers, c)
 }
 
+var printStack = sync.OnceFunc(func() {
+	fmt.Printf("All Goroutines Stack Trace:\n")
+	_ = pprof.Lookup("goroutine").WriteTo(os.Stdout, 1)
+})
+
 func (hc *HealthCheckImpl) broadcast(th *TabletHealth) {
 	hc.subMu.Lock()
 	defer hc.subMu.Unlock()
-	for c := range hc.subscribers {
+	for c, subscriber := range hc.subscribers {
 		select {
 		case c <- th:
 		default:
 			// If the channel is full, we drop the message.
 			hcChannelFullCounter.Add(1)
-			log.Warningf("HealthCheck broadcast channel is full, dropping message for %s", topotools.TabletIdent(th.Tablet))
+			log.Warningf("HealthCheck broadcast channel is full for %v, dropping message for %s", subscriber, topotools.TabletIdent(th.Tablet))
+			// Print the stack trace only once.
+			printStack()
 		}
```

Comment:
nit: this can be made part of `HealthCheckImpl` struct for adding a mock method for testing

Reply:
I added a test without adding it to the interface.